cmd/argo/commands/clustertemplate/root.go

```diff
@@ -0,0 +1,24 @@
+package clustertemplate
+
+import (
+	"github.com/spf13/cobra"
+)
+
+func NewClusterTemplateCommand() *cobra.Command {
+	var command = &cobra.Command{
+		Use:     "clustertemplate",
```

Comment:
I'm afraid that this will introduce even more confusion. We already have naming collisions with `WorkflowTemplate` vs `template`. Now we have to account for `clustertempalte`?

Reply:
Perhaps this CLI should really be under:

```
$ argo template cluster create [...]
```
or
```
$ argo template create --cluster [...]
```
?

Reply:
That was my initial thought to overload with workflowtemplate commands with flag, As per @alexec. The separate command will be more clean code and testable. I also agreed with that for code maintainability and testing.

Reply:
How about

```
$ argo cluster template
```
?

With this example it says symmetrical with `argo template` and allows us to extend it to things like `argo cluster cron` in the future.

I see Alex's point, but I think `argo clustertemplate` should not be used.

Reply:
> I see Alex's point, but I think `argo clustertemplate` should not be used.

Never mind any concerns about colliding names, it is too much to type and is very prone to typos.

Reply:
`argo cluster template` make sense